ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckOperation.java

```diff
@@ -75,7 +78,18 @@ public int execute() throws HiveException, IOException, TException, MetastoreExc
       MsckInfo msckInfo = new MsckInfo(SessionState.get().getCurrentCatalog(), tableName.getDb(), tableName.getTable(),
           desc.getFilterExp(), desc.getResFile(), desc.isRepairPartitions(),
           desc.isAddPartitions(), desc.isDropPartitions(), partitionExpirySeconds);
-      return msck.repair(msckInfo);
+      int result = msck.repair(msckInfo);
+      if (msckInfo.getSmallFilesStats() != null && !msckInfo.getSmallFilesStats().isEmpty()){
```

Comment:
can we log the warning in `msck.repair` while processing?

Reply:
I would prefer to not add the "small files warning" in `msck.repair()` function.
The purpose of repair() will be only plan and apply metadata fixes, user messaging can be belong to the check/plan stage and the CLI layer (MsckOperation).